### ISSUE TITLE
NewConfigFile not validating input

### ISSUE THREAD

**Issue by travertischio** [pulumi/pulumi-kubernetes#1283]
### Problem description

NewConfigFile is not validating the file path exists and is "successfully" creating no kubernetes resources

### Errors & Logs

I run pulumi up with the following code
```
        _, err := yaml.NewConfigFile(ctx, "developer-base", &yaml.ConfigFileArgs{
		File: "../../../rbac/developers.base.yaml",
	}, pulumi.Provider(k8sProvider))
	if err != nil {
		return err
	}
```
and I get
```
Previewing update (healthbyro/test):
     Type                                                           Name                        Plan       Info
     pulumi:pulumi:Stack                                            rodeo-eks-test
 +   └─ kubernetes:yaml:ConfigFile                                  developer-base              create
```        
so it says it wants to create the configfile, but no details of k8s resources
then I changed the file path in the pulumi code call to
```
        _, err := yaml.NewConfigFile(ctx, "developer-base", &yaml.ConfigFileArgs{
		File: "../rbac/developers.base.yaml",
	}, pulumi.Provider(k8sProvider))
	if err != nil {
		return err
	}
```
and got the same results when calling pulumi up

```
Previewing update (healthbyro/test):
     Type                                                           Name                        Plan       Info
     pulumi:pulumi:Stack                                            rodeo-eks-test
 +   └─ kubernetes:yaml:ConfigFile                                  developer-base              create
```      

### Affected product version(s)

2.8.2

### Reproducing the issue

Create a NewConfigFile with an incorrect path

### Suggestions for a fix

Validate that the file path actually exists.
Only "create" a config file if there are actual k8s resources being created

**Comment by lblackstone** [comment]
This is similar to #938

**Comment by EronWright** [comment]
This is fixed in pulumi-kubernetes v4.7.0 ([commit](https://github.com/pulumi/pulumi-kubernetes/pull/2709/files#diff-c877dcd2d0e8f6ba494bc6080f937b1a8a7e626c7f8806e3106777be8df242f7R168)).
